source/plugins/ruby/CAdvisorMetricsAPIClient.rb

```diff
@@ -311,6 +314,66 @@ def getInsightsMetrics(winNode: nil, metricTime: Time.now.utc.iso8601)
       return metricDataItems
     end
 
+    def getPersistentVolumeClaimMetrics(metricJSON, hostName, metricNameToCollect, metricNameToReturn, metricPollTime)
+      metricItems = []
+      clusterId = KubernetesApiClient.getClusterId
+      clusterName = KubernetesApiClient.getClusterName
+      begin
+        metricInfo = metricJSON
+        metricInfo["pods"].each do |pod|
+          podUid = pod["podRef"]["uid"]
+          podName = pod["podRef"]["name"]
+          podNamespace = pod["podRef"]["namespace"]
+
+          excludeNamespace = false
+          if (podNamespace.include? "kube-system") && @pvKubeSystemCollectionMetricsEnabled == "false"
+            excludeNamespace = true
+          end
+
+          if (!excludeNamespace && !pod["volume"].nil?)
+            pod["volume"].each do |volume|
+              if (!volume["pvcRef"].nil?)
+                pvcRef = volume["pvcRef"]
+                if (!pvcRef["name"].nil?)
+
+                  # A PVC exists on this volume
+                  pvcName = pvcRef["name"]
+                  pvName = volume["name"]
+
+                  metricItem = {}
+                  metricItem["CollectionTime"] = metricPollTime
+                  metricItem["Computer"] = hostName
```

Comment:
isn;t this the hostname for current node ? So for windows/remote, does it translate to that specific remote node's name ?